src/ui/file-browser/common/CreateTextFileForm.js

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm } from 'redux-form';
+import { Button, Row, Col, ControlLabel, Icon } from 'patternfly-react';
+import { formattedText, required, maxLength } from '@entando/utils';
+import { Link } from '@entando/router';
+import { FormattedMessage } from 'react-intl';
+import RenderTextAreaInput from 'ui/common/form/RenderTextAreaInput';
+import FormLabel from 'ui/common/form/FormLabel';
+import { ROUTE_FILE_BROWSER } from 'app-init/router';
+
+export const maxLength50 = maxLength(50);
+
+const RenderTextInput = (field) => {
+  const {
+    input, append, meta: { touched, error }, disabled,
+  } = field;
+  return (
+    <div className={`CreateTextFileForm__input ${(touched && error) ? 'form-group has-error' : 'form-group'}`}>
+      <input
+        {...input}
+        type="text"
+        id={input.name}
+        className="form-control CreateTextFileForm__input-text"
+        disabled={disabled}
+      />
+      {append && <span className="AppendedLabel">{append}</span>}
+      {touched && ((error && <span className="help-block">{error}</span>))}
+    </div>
+  );
+};
+
+
+export class CreateTextFileFormBody extends Component {
+  componentWillMount() {
+    const { filename, onWillMount } = this.props;
+    onWillMount(filename);
+  }
+  render() {
+    const {
+      invalid, submitting, handleSubmit, mode, filename, onClickDownload,
+    } = this.props;
+    const onSubmit = (ev) => {
+      ev.preventDefault();
+      handleSubmit();
+    };
```

Comment:
this is beyond unnecessary: handlesubmit already prevents the default. you should just have handle submit as the form onSubmit property:

` <form onSubmit={handleSubmit} >` there is no need for any of this additional stuff.